### PR TITLE
Breaks up gameplay namespace, leverages events, other small changes

### PR DESCRIPTION
**Pull request by williamhammond**
I should have moved everything around in a separate PR 🙈. 

Changes
1. I added a "Development" repo and added it to the .gitignore. I figured people use this to work on random test maps
2. I made more specific namespaces under scripts since basically everything is "Gameplay"
3. Decoupled the TurnController from Player by using C#'s event system. Basically instead of needing a reference to the player controller we publish an event and the TurnController adds a player to the list. The main reason I did this is because I was trying to test the player without any of the controller logic and it exploded during startup. Also I'm thinking we should use this https://mirror-networking.com/ for networking, and the event system works really well with it, so we'll probably end up needing to add events to everything anyway

I made some other small changes mostly based off of a unity udemy class, i'll call them out in-line

### CODE REVIEW

File: Assets/Scripts/UI/CurrentTurnText.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using TMPro;
+using UnityEngine;
+
+namespace UI
```
Comment (williamhammond) [inline Assets/Scripts/UI/CurrentTurnText.cs:4]: original https://github.com/williamhammond/NotWorms/blob/4b9988f666f1cc0caa48826b9078b2cf88afa9eb/Assets/Scripts/Gameplay/CurrentTurnText.cs

File: Assets/Scripts/UI/CurrentTurnText.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public class CurrentTurnText : MonoBehaviour
+    {
+        [SerializeField]
+        private TMP_Text labelText;
```
Comment (williamhammond) [inline Assets/Scripts/UI/CurrentTurnText.cs:9]: I'm not sure if this is technically more correct than `TextMeshProUGUI`, but that's what the teacher in my unity class did so i copied him

File: Assets/Scripts/UI/CurrentTurnText.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+        public void UpdateTurn(int input)
+        {
+            labelText.text = $"{input}";
```
Comment (williamhammond) [inline Assets/Scripts/UI/CurrentTurnText.cs:13]: I think the templated string like this is more idimoatic

File: Assets/Scripts/UI/CurrentTurnText.cs (hunk 2 above)
Comment (saunders-d) [reply Assets/Scripts/UI/CurrentTurnText.cs:9]: TextMeshProUGUI is used when working with canvas elements. TextMeshPro is used for meshes in 3D world space.

Comment (xhocquet) [reply Assets/Scripts/UI/CurrentTurnText.cs:9]: It was setup with a canvas so it should use `TextMeshProUGUI ` right?

**Comment by williamhammond** [comment]
I'm going to break this up into smaller PRs

**Pull request closed without merging**